### PR TITLE
Clarify labels and tokens

### LINKED ISSUE #995: Clarify process for adding labels in the token reference
The token reference relies on entity labels for its examples, but doesn't indicate how to add labels to agents using the agent config file. Update the token reference to make this process more clear.

### LINKED ISSUE #999: Use underscores in token substitution examples
It turns out that dashes don't (and can't) work in token substitution. We should change the examples.

### PR DESCRIPTION
**Pull request by apaskulin**
## Description

This PR addresses some confusion around labels and tokens.

- Changes the reference docs for each type to specify character limits for label keys
- Adds a section about managing entity labels to the entity reference
- Updates the token reference to link to the new section in the entity reference
- Makes a few stylistic and formatting changes

## Motivation and Context

Closes #995 
Closes #999

### CODE REVIEW

File: content/sensu-go/5.0/reference/entities.md [hunk 1]
```diff
@@ -10,25 +10,112 @@ menu:
...
 
-## Proxy Entities
+An entity represents anything (ex: server, container, network switch) that needs to be monitored, including the full range of infrastructure, runtime and application types that compose a complete monitoring environment (from server hardware to serverless functions).
+We call these monitored parts of an infrastructure "entities".
+An entity not only provides context to event data (what/where the event is from) but an event's uniqueness is determined by the check name and the ID of the entity upon which the check ran.
```
Comment (echlebek) [inline content/sensu-go/5.0/reference/entities.md]: We recently transitioned from "ID" to "name" for entities, as of GA. We should avoid mentioning ID in the context of entities in the future.

**Review by echlebek — CHANGES_REQUESTED** [review]
I think the changes from
```
us-west-2
```
to
```
us_west2
```
can be reverted, as label values can contain any valid UTF-8 string.

**Comment by echlebek** [comment]
Looks good in general, the only thing we need to change is any mention of "ID". Before GA, we decided we would use only "name" and not use "ID" anywhere in the project.

File: content/sensu-go/5.0/reference/entities.md (hunk 1 above)
Comment (apaskulin) [inline content/sensu-go/5.0/reference/entities.md]:
```suggestion
An entity not only provides context to event data (what/where the event is from) but an event's uniqueness is determined by the check name and the name of the entity upon which the check ran.
```

**Review by echlebek — APPROVED** [review]

**Comment by apaskulin** [comment]
Thanks, @echlebek! I've updated ID to name, and reverted the change to us-west-x.

**Pull request merged**